GoMeet/src/test/java/com/google/sps/LocationServletTest.java

```diff
@@ -75,51 +76,35 @@ public void tearDown() {
   }
 
   /** 
-   * Tests if new location entity is accurately added to the Database.
-   * Run this test twice to prove we're not leaking any state across tests.
+   * Tests if the recieved requests are correctly sent to the LocationDAO
+   * and the key is sent in the response.
    */
-  private void doPostTest() throws IOException {
-    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+  @Test
+  public void doPostTest() throws IOException {
+    // Set up request mock
+    when(request.getParameter("title")).thenReturn(LOCATION_A.getTitle());
+    when(request.getParameter("lat")).thenReturn(Double.toString(LOCATION_A.getLat()));
+    when(request.getParameter("lng")).thenReturn(Double.toString(LOCATION_A.getLng()));
+    when(request.getParameter("note")).thenReturn(LOCATION_A.getNote());
    
-    when(request.getParameter("title")).thenReturn(TITLE_A);
-    when(request.getParameter("lat")).thenReturn(LAT_A);
-    when(request.getParameter("lng")).thenReturn(LNG_A);
-    when(request.getParameter("note")).thenReturn(NOTE_A);
-
-    StringWriter stringWriter = new StringWriter();
-    PrintWriter writer = new PrintWriter(stringWriter);
-    when(response.getWriter()).thenReturn(writer);
+    // Set up DAO mock
+    String keyString = KeyFactory.keyToString(KeyFactory.createKey("Location", 123));
+    when(mockedLocationDao.save((Location)notNull())).thenReturn(keyString);
 
-    new LocationServlet().doPost(request, response);
+    LocationServlet servlet = new LocationServlet();
+    servlet.setDao(mockedLocationDao);
+    servlet.doPost(request, response);
     
-    // Check location entity was added to Datastore
-    assertEquals(1, ds.prepare(new Query("Location")).countEntities(withLimit(10)));
-
-    Query query = new Query("Location");
-    PreparedQuery preparedQuery = ds.prepare(query);
-    Entity result = preparedQuery.asSingleEntity();
-
-    // Check the entity property values were assigned correctly
-    assertEquals(TITLE_A, result.getProperty("title"));
-    assertEquals(LAT_A_VALUE, result.getProperty("lat"));
-    assertEquals(LNG_A_VALUE, result.getProperty("lng"));
-    assertEquals(NOTE_A, result.getProperty("note"));
-    assertEquals(INIT_VOTE_COUNT, ((Long) result.getProperty("voteCount")).intValue());
+    // Check the mockedDao was called with the correct parameters
+    ArgumentCaptor<Location> captor = ArgumentCaptor.forClass(Location.class);
+    verify(mockedLocationDao, times(1)).save(captor.capture());
```

Comment:
Just to confirm, your test is checking to see if LocationDao::save() is called once with the argument LOCATION_A?

Reply:
Yes, that is correct. And then the next assertEquals is to check that the response was sent with the keyString.